js/apps/events-page/dataServices/eventsService.js

```diff
@@ -0,0 +1,54 @@
+((app) => {
+	const eventsService = (CONSTANTS, $http, $q) => {
+		const isEventOnDate = (eventItem, eventDate) => {
+			const eventItemDate = (new Date(eventItem.EventStart)).toLocaleDateString();
+			return eventItemDate === eventDate;
+		};
+
+		const dateSplitter = (eventData) => {
+			let eventsByDate = [];
+			let lastEventDate;
+
+			angular.forEach(eventData, (eventItem) => {
```

Comment:
Should this break? If so maybe a ```for``` loop would be more appropriate?

Reply:
It should not break; I need to look at every event to figure out the date boundaries.